src/rgw/rgw_auth.h

```diff
@@ -64,6 +63,10 @@ class Identity {
   }
 
   virtual void to_str(std::ostream& out) const = 0;
+
+  virtual const boost::optional<std::string> get_subuser_name() const {
```

Comment:
1. I expect `get_subuser` would return the `RGWSubUser` ?  Maybe `get_subuser_name` is more explicit
2. Added const references